ethereum/eip712/eip712_test.go

```diff
@@ -0,0 +1,309 @@
+package eip712_test
+
+import (
+	"testing"
+
+	"cosmossdk.io/math"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/evmos/ethermint/ethereum/eip712"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/evmos/ethermint/crypto/ethsecp256k1"
+
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/evmos/ethermint/app"
+	"github.com/evmos/ethermint/encoding"
+
+	txtypes "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+
+	distributiontypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// Tests single-signer EIP-712 signature verification. Multi-signer verification tests are included
```

Comment:
Why separate these tests? Looks like this file includes integration tests which could be moved to the multi-signer tests. 

I'd include only isolated unit tests under `ethereum/eip712/`

Reply:
These aren't integration tests, they're unit tests to test `eip712/encoding` and `ethsecp256k1` logic.

Reply:
nit, if they are unit tests, then I'd split them up for individual functions, e.g.  `TestGetEIP712HashForMsg`